CONTRIBUTING.md

```diff
@@ -1,50 +1,105 @@
+we have designed it to help you clarify your idea.
 
-If you want to submit a new issue - whether it's a bug report or a request for a new feature - there are a few steps you should try first:
+## Advanced contributions
 
-1. Search the issue tracker to see if someone else has already submitted this issue (see the links above)
-2. Double-check to make sure it's a genuine issue: if it's a bug, can you reproduce it? If it's a feature request, does it make sense in the context of Unlock?
+If you are ready to go a step further, and contribute changes to the codebase, then we ask for more rigorous standards.
 
-If you're sure you need to create a new issue, please make a best effort to follow the templates we've created. Be as specific as possible, including by using a clear, descriptive title for the issue. This allows contributors to address the issue you've uncovered with the most possible context.
+To contribute a design idea, copy-editing, or other contributions not directly related to the functioning code, please
+open an issue as described above containing your contribution.
 
-#### Fixing an existing issue
+If you want to contribute code, then please read the steps below on how to create a local branch on your fork, and create a
+pull request from that branch.
 
-When you're ready to contribute code, first check the following:
+### Contributing code: Addressing an existing issue
 
-1. Your code is well-documented
-2. Your classes, functions, interfaces, etc are all clearly-named
-3. Your style and naming conventions adhere to the style used in the rest of the codebase
-4. You have included JavaScript tests where appropriate in the `/unlock-app/src/__tests__` or `smart-contracts/test` folders for the Unlock app and smart contracts respectively
+First off, thank you! We recognize the effort it takes to contribute code, and value both your code and your time.
 
-Commit the code in a descriptive branch with a name that represents your contribution, in the form `yourname-short-title`.
+When you are ready to contribute code, please ensure that it is a direct implementation of a solution to a problem
+mentioned in an [existing issue on our issue tracker](https://github.com/unlock-protocol/unlock/issues?q=is%3Aissue+is%3Aopen+label%3Abug). If you wish to tackle an issue, please ask by commenting on the issue, we will get back to you very quickly.
 
-Finally, submit a pull request, with the following rules:
+When you are ready to contribute to the codebase, then you will need to create a [pull request](https://help.github.com/articles/about-pull-requests/) from a local [fork](https://help.github.com/articles/fork-a-repo/)
+of the [Unlock Protocol git repository](https://github.com/unlock-protocol/unlock). We strongly recommend that you
+do all development on a [branch](https://help.github.com/articles/about-branches/) named with this convention:
+`yourname-short-title`. where `yourname` is your github username.
```

Comment:
```suggestion
`yourname-short-title-issuenumber`. where `yourname` is your github username, and `issuenumber` is the number of the issue your contribution addresses.
```

Reply:
This is complex enough I'm also going to add an example branch name